### PR TITLE
Adding Agent

### PR DESCRIPTION
**Pull request by cam156**
## Description

Added model, controller, and views to match with the data dictionary

Added the basics for the rest of the Agent work.

Connected to #730

### CODE REVIEW

**Comment by awead** [comment]
@cam156 If you rebase this against master, we'll get code coverage reports back with Code Climate.

File: app/cho/agent/resources_controller.rb [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    # GET /agents/1/edit
+    def edit
+      @agent = load_change_set
+    end
+
+    # POST /agents
+    # POST /agents.json
+    def create
+      validate_save_and_respond(initialize_change_set, :new)
```
[the remark this answers is not in the document]
Comment (cam156) [reply app/cho/agent/resources_controller.rb]: I assume we will want to search agents.  That was the point of putting them into the system.  That was why I indexed them.

**Comment by cam156** [comment]
@awead we are already getting code climate reports.  Not sure why you are asking for a rebase.

File: app/views/agent/resources/_agent.json.jbuilder [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# frozen_string_literal: true
```
[the remark this answers is not in the document]
Comment (cam156) [reply app/views/agent/resources/_agent.json.jbuilder]: You do not want json format?  I just copied from data dictionary, but we can deleted that.

[the remark this answers is not in the document]
Comment (awead) [reply app/views/agent/resources/_agent.json.jbuilder]: There isn't an immediate need for them, but I guess leaving them in won't hurt.

**Pull request merged**